### PR TITLE
Usage stats: Divide collection into multiple functions to isolate failures

### PR DESCRIPTION
**Pull request by sakjur**
**What this PR does / why we need it**:

This PR improves the resiliency of the usage stats report by creating several collectors that each are capable of failing in at most one way. In addition, there are minor improvements to debugging of usage stats themselves and the Prometheus flavor detection specifically.

**Which issue(s) this PR fixes**:

Reduces effect of grafana/grafana-enterprise#3373

**Special notes for your reviewer**:

The diff looks more intimidating than it actually is. I have not deleted any metrics, and only added two new ones (plus a Prometheus flavor for HTTP network _errors_ [defined here as when there's no HTTP server, so a 4xx/5xx wouldn't be an error here]). Almost all of the "removed" code is still there.

### CODE REVIEW

**Comment by DanCech** [comment]
Code looks good, we could potentially split `collectSystemStats` into the stats that rely on the database and those that are derived from config but I don't think there's much benefit to doing so.

**Review by ying-jeanne — APPROVED** [review]
Nit refactor LGTM. i saw log.Error in the err, wonder when the gatherMetrics is called, maybe they could investigate on the logs to see what is happening for those using usage-stats(if it is HG clients...)

File: pkg/infra/usagestats/statscollector/service_test.go [hunk 1]
```diff
@@ -216,6 +267,31 @@ func TestCollectingUsageStats(t *testing.T) {
 		},
 	}
 
+	{
+		db, err := s.collectDatasourceStats(context.Background())
+		require.NoError(t, err)
+
+		assert.EqualValues(t, 9, db["stats.ds."+models.DS_ES+".count"])
+		assert.EqualValues(t, 10, db["stats.ds."+models.DS_PROMETHEUS+".count"])
+		assert.EqualValues(t, 11+12, db["stats.ds.other.count"])
```
[the remark this answers is not in the document]
Comment (sakjur) [reply pkg/infra/usagestats/statscollector/service_test.go:276]: 🔥 Nice one 😂

> Ten. No, nine. No, wait, eight. I mean eight.

**Review by DanCech — APPROVED** [review]

**Comment by sakjur** [comment]
> Code looks good, we could potentially split collectSystemStats into the stats that rely on the database and those that are derived from config but I don't think there's much benefit to doing so.

Yup, I'll leave that to if we have errors with the SQL call 🙂 This should already tell us which collectors are less stable than others (I don't think it will be that one)

**Pull request merged**